src/Microsoft.Extensions.Http/Logging/LoggingScopeHttpMessageHandler.cs

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Extensions.Http.Logging
+{
+    public class LoggingScopeHttpMessageHandler : DelegatingHandler
+    {
+        private ILogger _logger;
+
+        public LoggingScopeHttpMessageHandler(ILogger logger)
+        {
+            if (logger == null)
+            {
+                throw new ArgumentNullException(nameof(logger));
+            }
+
+            _logger = logger;
+        }
+
+        protected async override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            if (request == null)
+            {
+                throw new ArgumentNullException(nameof(request));
+            }
+
+            using (_logger.BeginScope<Scope>(new Scope(request)))
+            {
+                _logger.LogInformation("Processing Request {HttpMethod} to {Uri}", request.Method, request.RequestUri.AbsoluteUri);
+                var response = await base.SendAsync(request, cancellationToken);
+                return response;
+            }
+
+
+                
+        }
+
+        private struct Scope : IReadOnlyList<KeyValuePair<string, object>>
```

Comment:
I started actually implementing logging, but decided that I want to get feedback on the options/startup thing first. I'll want to solicit feedback about diagnostics next.